### PR TITLE
chore(recipe):Add a recipe for using FontAwesome

### PR DESCRIPTION
**Pull request by laurieontech**
## Description

Adding a recipe for using fontawesome in a Gatsby site.

## Related Issues

Addresses #17703

### CODE REVIEW

**Comment by marcysutton** [comment]
Until we do the engineering work to make the site automatically list recipes, this one needs to be added to the recipes overview page. Here are some tips in the contributing docs: https://www.gatsbyjs.org/contributing/docs-templates/#recipes

File: docs/docs/recipes/styling-css.md [hunk 1]
```diff
@@ -415,3 +415,44 @@ _NOTE: So for the above example, the relevant CSS declaration would be `font-fam
 - [Typography.js](/docs/typography-js/) - Another option for using Google fonts on a Gatsby site
 - [The Typefaces Project Docs](https://github.com/KyleAMathews/typefaces/blob/master/README.md)
 - [Live example on Kyle Mathews' blog](https://www.bricolage.io/typefaces-easiest-way-to-self-host-fonts/)
+
+## Using FontAwesome
+
+Using [FontAwesome](https://fontawesome.com/) gives you access to thousands of icons for use on your site. Since Gatsby sites are React sites, we recommend you use the [react-fontawesome](https://github.com/FortAwesome/react-fontawesome) SVG library.
```
Comment (marcysutton) [inline docs/docs/recipes/styling-css.md]:
```suggestion
Using [FontAwesome](https://fontawesome.com/) gives you access to thousands of icons for use on your site. Since Gatsby sites are React sites, it's recommended to use the [react-fontawesome](https://github.com/FortAwesome/react-fontawesome) SVG library.
```

File: docs/docs/recipes/styling-css.md [hunk 2]
```diff
@@ -415,3 +415,44 @@ _NOTE: So for the above example, the relevant CSS declaration would be `font-fam
...
+
+### Prerequisites
+
+- A [Gatsby site](/docs/quick-start)
+- The [Gatsby CLI](/docs/gatsby-cli/) installed
```
Comment (marcysutton) [inline docs/docs/recipes/styling-css.md]: I feel like the order of these should be flipped, since you need the Gatsby CLI to build a Gatsby site. (looks like this is listed in that order on other recipes too, which should be updated..but that could be a separate PR)

File: docs/docs/recipes/styling-css.md [hunk 3]
```diff
@@ -415,3 +415,44 @@ _NOTE: So for the above example, the relevant CSS declaration would be `font-fam
...
+
+### Directions
+
+1. Install the react-fontawesome dependencies.
```
Comment (marcysutton) [inline docs/docs/recipes/styling-css.md]:
```suggestion
1. Install the `react-fontawesome` dependencies.
```

File: docs/docs/recipes/styling-css.md [hunk 4]
```diff
@@ -415,3 +415,44 @@ _NOTE: So for the above example, the relevant CSS declaration would be `font-fam
...
+
+```shell
+npm install @fortawesome/fontawesome-svg-core  @fortawesome/free-brands-svg-icons @fortawesome/react-fontawesome
+```
+
+> Note that there are multiple icon librarys within `react-fontawesome`. You may also be interested in `free-regular-svg-icons` and `free-solid-svg-icons` which you would install the same way.
```
Comment (marcysutton) [inline docs/docs/recipes/styling-css.md]:
```suggestion
> Note that there are multiple icon libraries within `react-fontawesome`. You may also be interested in `free-regular-svg-icons` and `free-solid-svg-icons` which you would install the same way.
```

File: docs/docs/recipes/styling-css.md [hunk 5]
```diff
@@ -415,3 +415,44 @@ _NOTE: So for the above example, the relevant CSS declaration would be `font-fam
...
+
+2. Import the `FontAwesomeIcon` component and the icon you want to use. Then use the icon as a component directly in your JSX.
```
Comment (marcysutton) [inline docs/docs/recipes/styling-css.md]:
```suggestion
2. Import the `FontAwesomeIcon` component and the icon you want to use. Then use the icon as a component directly in your JSX files:
```

File: docs/docs/recipes/styling-css.md [hunk 6]
```diff
@@ -415,3 +415,44 @@ _NOTE: So for the above example, the relevant CSS declaration would be `font-fam
...
+const IndexPage = () => (
+  <Layout>
+    <FontAwesomeIcon icon={faReact} /> //highlight-line
+  </Layout>
+)
+
+export default IndexPage
+```
+
+> This example imports a specific icon and uses it for improved performance. As an alternative you can [import the icons and build a library](https://github.com/FortAwesome/react-fontawesome#build-a-library-to-reference-icons-throughout-your-app-more-conveniently).
```
Comment (marcysutton) [inline docs/docs/recipes/styling-css.md]:
```suggestion
> This example imports a single, specific icon and uses it for improved performance. As an alternative, you can [import the icons and build a library](https://github.com/FortAwesome/react-fontawesome#build-a-library-to-reference-icons-throughout-your-app-more-conveniently).
```

**Comment by ehowey** [comment]
You can completely ignore this - seems like this PR is basically done. Was there any conversation about using [react-icons](https://react-icons.netlify.com/#/)?

Gives easy access to FA and a whole host of other icon libraries like material design and feather icons...The API is also pretty easy from my experience working with it.  Just a thought, as I said this looks done and great.  I was just looking through PRs and saw this.

**Comment by laurieontech** [comment]
There was. In the docs as a whole, there is prior art for using react-fontawesome so I went with that approach. Both are reasonable options.
